attributioncode/validator.go

```diff
@@ -5,18 +5,30 @@ import (
 	"crypto/sha256"
 	"encoding/base64"
 	"encoding/hex"
+	"fmt"
 	"net/url"
 	"time"
 
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
+const maxUnescapedCodeLen = 600
+
 var validAttributionKeys = map[string]bool{
-	"source":   true,
-	"medium":   true,
-	"campaign": true,
-	"content":  true,
+	"source":     true,
+	"medium":     true,
+	"campaign":   true,
+	"content":    true,
+	"experiment": true,
+	"variation":  true,
```

Comment:
Just noting, and we probably don't need to change anything here -- this allows experiment or variation to be set without the other, which could totally be valid, e.g., some experiment with no variation compared against a baseline no-experiment.

Reply:
Oh, does this need "ua"?  https://searchfox.org/mozilla-central/rev/a92ed79b0bc746159fc31af1586adbfa9e45e264/browser/components/attribution/AttributionCode.jsm#35

Probably could use a comment in the code for validAttributionKeys that it should match what is in firefox also.